package haptix.comm.msgs;

/// \ingroup haptix_comm_msgs
/// \interface hxSensor
/// \brief Sensor message. This message specifies the sensor information gained
/// in a simulation update.

import "hxQuaternion.proto";
import "hxTime.proto";
import "hxVector3.proto";

message hxSensor
{
  repeated float motor_pos              = 1 [packed=true];
  repeated float motor_vel              = 2 [packed=true];
  repeated float motor_torque           = 3 [packed=true];
  repeated float joint_pos              = 4 [packed=true];
  repeated float joint_vel              = 5 [packed=true];
  repeated float contact                = 6 [packed=true];
  repeated hxVector3 imu_linear_acc     = 7;
  repeated hxVector3 imu_angular_vel    = 8;
  repeated hxQuaternion imu_orientation = 9;
  required hxTime time_stamp            = 10;
}
